tests/test_universal_pipeline.py

```diff
@@ -151,6 +151,12 @@ def body(self):
     result = response.context["deserialized_data"]
     assert result["success"] is True
 
+    # Simple JSON with complex content_type
+    response = build_response(b'{"success": true}', content_type="application/vnd.microsoft.appconfig.kv+json")
+    raw_deserializer.on_response(None, response, stream=False)
+    result = response.context["deserialized_data"]
+    assert result["success"] is True
+
     # JSON with UTF-8 BOM
     response = build_response(b'\xef\xbb\xbf{"success": true}', content_type="application/json; charset=utf-8")
```

Comment:
Isn't adding a BOM in violation of https://tools.ietf.org/html/rfc7158#section-8.1?

Reply:
https://github.com/Azure/msrest-for-python/issues/144